### ISSUE TITLE
The launcher for Linux/Mac

### ISSUE THREAD

**Issue by gbudny** [rwengine/openrw#77]
Hi

Do you plan to create a launcher that will help you to indicate the location of the data from Grand Theft Auto III?

I could be also used to change other settings e.g.:

"Options:
-w n, -h n sets initial window size
--newgame starts a new game automatically"

**Comment by tsjost** [comment]
Yes, some kind of launcher will be needed, but we haven't begun thinking about how to do that yet.

**Comment by PureTryOut** [comment]
I honestly don't think this is needed. A simple popup asking where the gamefiles are located would be enough. Resolution settings would be done entirely in-game. First time window size could be something that fit on every screen, 800x600 or something.

Maybe the gamefiles popup could be in the game window itself, before any menu is loaded? This probably needs to be done later then though, as like @danhedron said on IRC, there aren't any UI elements to speak of right now.

**Comment by gbudny** [comment]
The authors of Freeablo created a launcher that works on Linux and Mac:

https://github.com/wheybags/freeablo/pull/217

Perhaps, you can do something similar because the location of the data from Grand Theft Auto III could be in different places. Some people can install this game from a CD when they use Wine, but they are also two digital versions from Steam and App Store.

**Comment by PureTryOut** [comment]
Like I said, an in-game window would be preferred, as this would even allow controller usage, and fits in better with the game.

**Comment by danhedron** [comment]
Doing this in-game would allow for controller support much more easily compared to using Qt, so that's what I have in mind.

**Comment by gbudny** [comment]
That is the interesting idea to solve this issue.

**Comment by JayFoxRox** [comment]
@danhedron can't really be done in the menu because the menu requires fonts to be loaded.

How is this even an issue?

**Just load the path from config**
(Maybe) optionally have a CLI to create a new config by searching the standard path either by checking the registry on windows or looking in the current folder (considering this like a patched/replacement gta3.exe). On unix we'd do it the same way but load GTA from any of the standard paths: "/opt/gta3" etc.

If anybody wants a launcher just have them have a launcher, they can just export a config file (which specifys the game path) and launch OpenRW directly.
Wether there is another CLI to prevent certain menu options to appear in the ingame-menu (due to ambiguity of launcher and menu settings) is a different issue.
